core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -42,32 +49,39 @@
 
    private static final Flag[] EMPTY_FLAGS = new Flag[0];
    private final long flags;
+   private final Object lockOwner;
    private final CacheImpl<K, V> cacheImplementation;
 
    public DecoratedCache(AdvancedCache<K, V> delegate) {
       this(delegate, EMPTY_FLAGS);
    }
 
    public DecoratedCache(AdvancedCache<K, V> delegate, Flag... flags) {
-      super(delegate);
+      this(delegate, null, flags);
+   }
 
+   public DecoratedCache(AdvancedCache<K, V> delegate, Object lockOwner, Flag... flags) {
+      super(delegate);
       if (flags == null)
-         throw new IllegalArgumentException("There is no point in using a DecoratedCache without specifying any Flags.");
+         throw new IllegalArgumentException("There is no point in using a DecoratedCache if Flags are set.");
```

Comment:
Yeah this was another conflict resolution one that I missed. I fixed the original merge but the comment wasn't updated. I can change the logic though as the previous commit is incorrect as you mentioned.